arches_references/management/commands/controlled_lists.py

```diff
@@ -0,0 +1,120 @@
+            nargs="*",
+            required=True,
+            help="One or more collections to migrate to controlled lists",
+        )
+
+        parser.add_argument(
+            "-ho",
+            "--host",
+            action="store",
+            dest="host",
+            default="http://localhost:8000/plugins/controlled-list-manager/item/",
+            help="Provide a host for URI generation. Default is localhost",
+        )
+
+        parser.add_argument(
+            "-ow",
+            "--overwrite",
+            action="store_true",
+            dest="overwrite",
+            default=False,
+            help="Overwrite the entire controlled list and its list items/values. Default false.",
+        )
+
+        parser.add_argument(
+            "-psl",
+            "--preferred_sort_language",
+            action="store",
+            dest="preferred_sort_language",
+            default="en",
+            help="The language to use for sorting preferred labels. Default 'en'",
+        )
+
+    def handle(self, *args, **options):
+        if options["operation"] == "migrate_collections_to_controlled_lists":
+            self.migrate_collections_to_controlled_lists(
+                collections_to_migrate=options["collections_to_migrate"],
+                host=options["host"],
+                overwrite=options["overwrite"],
+                preferred_sort_language=options["preferred_sort_language"],
+            )
+
+    def migrate_collections_to_controlled_lists(
+        self,
+        collections_to_migrate,
+        host,
+        overwrite,
+        preferred_sort_language,
+    ):
+        """
+        Uses a postgres function to migrate collections to controlled lists
+
+        Example usage:
+            python manage.py controlled_lists
+                -o migrate_collections_to_controlled_lists
+                -co 'Johns list' 'Getty AAT'
+                -ho 'http://localhost:8000/plugins/controlled-list-manager/item/'
+                -psl 'fr'
+                -ow
+        """
+
+        collections_in_db = list(
+            Value.objects.filter(
+                value__in=collections_to_migrate,
+                valuetype__in=["prefLabel", "identifier"],
+                concept__nodetype="Collection",
+            ).values_list("value", flat=True)
+        )
+
+        failed_collections = [
+            collection
+            for collection in collections_to_migrate
+            if collection not in collections_in_db
+        ]
+
+        if len(failed_collections) > 0:
+            self.stdout.write(
+                "Failed to find the following collections in the database: %s"
+                % ", ".join(failed_collections)
+            )
+
+        if len(collections_in_db) > 0:
+            from django.db import connection
+
+            cursor = connection.cursor()
+            cursor.execute(
+                """
+                select * from __arches_migrate_collections_to_clm(
```

Comment:
Cool. In the future we can consider implementing the `Func` [expressions API](https://docs.djangoproject.com/en/5.0/ref/models/expressions/#func-expressions) to be able to use custom db functions with the ORM.